### PR TITLE
BREAKING: ERC1155 support

### PR DESCRIPTION
**Pull request by gantunesr**
## Description

Currently we don't support the collectible standard ERC-1155, the focus of this development is to integrated the ERC-1155 ABI to access the appropriate methods of the smart contract.

## Breaking Changes

- **CollectiblesController**: Consumers of the `CollectiblesController` will have to update the creation parameters and initialization of the object created. The parameters are:
  - AssetsContractController.getOwnerOf
  -  AssetsContractController.balanceOfERC1155Collectible
  - AssetsContractController.uriERC1155Collectible

## To Do

- [x] Add ERC-1155 ABI dependence.
- [x] Develop classes to represent ERC-721 and ERC-1155.
- [x] Update the AssetsContractController to handle the new standard.
- [x] Update the CollectiblesController to verify the ownership of any of the mentioned standards.
- [x] Update testing and docs.

## Next Steps

- Develop class to represent ERC-20
- Move the smart contracts ABI (ERC-20, ERC-721, and ERC-1155) to a MM repository.

### CODE REVIEW

File: src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    return this.contractSupportsInterface(
+      contract,
+      ERC1155_METADATA_URI_INTERFACE_ID,
+    );
+  };
+
+  /**
+   * Query if contract implements ERC721Enumerable interface.
+   *
+   * @param contract - ERC721 asset contract.
```
Comment (sethkfman) [inline src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts:23]: Update comment

File: src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
+const ERC1155_METADATA_URI_INTERFACE_ID = '0x0e89341c';
+const ERC1155_TOKEN_RECEIVER_INTERFACE_ID = '0x4e2312e0';
+
+export class ERC1155Standard {
+  /**
+   * Query if contract implements ERC721Metadata interface.
+   *
+   * @param contract - ERC721 asset contract.
```
Comment (sethkfman) [inline src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts:8]: Update comment

File: src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts (hunk 1 above)
Comment (gantunesr) [reply src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts]: Done

File: src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts (hunk 2 above)
Comment (gantunesr) [reply src/assets/CollectibleStandards/ERC1155/ERC1155Standard.ts]: Done

File: src/assets/CollectiblesController.ts [hunk 3]
```diff
@@ -729,6 +745,49 @@ export class CollectiblesController extends BaseController<
...
+    try {
+      const balance = await this.balanceOfERC1155Collectible(
+        ownerAddress,
+        collectibleAddress,
+        collectibleId,
+      );
+      return balance > 0;
+      // eslint-disable-next-line no-empty
+    } catch {
+      // Ignore ERC-1155 contract error
```
[the remark this answers is not in the document]
Comment (sethkfman) [reply src/assets/CollectiblesController.ts:780]: I see your reasoning.

**Pull request merged**